### ISSUE TITLE
net: socket: register: net.socket.register.tls fails

### ISSUE THREAD

**Issue by katgiadla** [zephyrproject-rtos/zephyr#73337]
**Describe the bug**
The test `net/socket/register/net.socket.register.tls` fails.

Observed for:
- `nrf5340dk/nrf5340/cpuapp/ns`
- `nrf9160dk/nrf9160/ns`

**To Reproduce**
1. have `nrf9160dk/nrf9160/ns` connected
2. go to your zephyr dir
3. call `./scripts/twister -T tests/net/socket/register -p nrf9160dk/nrf9160/ns --device-testing --device-serial /dev/ttyACM0 -v --inline-logs`
4. See console output with error

**Expected behavior**
Valid console output

**Impact**
Not clear

**Logs and console output**
Logs from `build.log`: 

```
arm-zephyr-eabi/bin/ld.bfd: zephyr/zephyr_pre0.elf section `rodata' will not fit in region `FLASH'
arm-zephyr-eabi/bin/ld.bfd: region `FLASH' overflowed by 25716 bytes
arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj): in function `ssl_update_checksum_sha256':
/zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:944: undefined reference to `mbedtls_md_error_from_psa'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj): in function `ssl_update_checksum_sha384':
zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:957: undefined reference to `mbedtls_md_error_from_psa'
arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj): in function `ssl_calc_finished_tls_generic':
zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:8400: undefined reference to `mbedtls_md_error_from_psa'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj): in function `ssl_calc_verify_tls_psa':
zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:7314: undefined reference to `mbedtls_md_error_from_psa'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj): in function `ssl_update_checksum_start':
zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:927: undefined reference to `mbedtls_md_error_from_psa'
opt/zephyr-sdk/arm-zephyr-eabi/bin/lib/gcc/arm-zephyr-eabi/12.2.0/arm-zephyr-eabi/bin/ld.bfd: modules/mbedtls/libmodules__mbedtls.a(ssl_tls.c.obj):zephyr/zephyr/twister-out/nrf5340dk_nrf5340_cpuapp_ns/tests/net/socket/register/net.socket.register.tls/modules/crypto/mbedtls/library/ssl_tls.c:876: more undefined references to `mbedtls_md_error_from_psa' follow
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information)**
- OS: Ubuntu 20.04.1 LTS
- Toolchain: Zephyr SDK 0.16.5-1
- Commit SHA or Version used: `v3.6.0-4561-g8c8e2f6243f1`

**Comment by jukkar** [comment]
This is a regression after #72243 is merged. @valeriosetti could you take a look why the memory consumption increased and we have undefined symbols in the compilation?

**Comment by valeriosetti** [comment]
Sure! Meanwhile can you assign me this issue? So it will be easier to find this in case I accidentally close the page

**Comment by katgiadla** [comment]
> Sure! Meanwhile can you assign me this issue? So it will be easier to find this in case I accidentally close the page

I cannot do that. @jukkar, could you assign him?

**Comment by valeriosetti** [comment]
I have pushed a fix in #73364. Can you please take a look and let me know it that's OK for you?

**Comment by katgiadla** [comment]
> I have pushed a fix in #73364. Can you please take a look and let me know it that's OK for you?

it works on-target.
